cmds/mount/mount.go

```diff
@@ -48,7 +75,11 @@ func main() {
 	if *ro {
 		flags |= unix.MS_RDONLY
 	}
-	if err := unix.Mount(a[0], a[1], *fsType, flags, strings.Join(data, ",")); err != nil {
+	if *fsType == "" {
+		log.Printf("No file system type provided!\n")
+		log.Fatalf("Usage: mount [-r] [-o mount options] -t fstype dev path")
+	} else if err := unix.Mount(a[0], a[1], *fsType, flags, strings.Join(data, ",")); err != nil {
```

Comment:
The problem here is that there can be lots of reasons for an error on mount, and this is going to confuse people if they're told that it's a nonexistent file system if that's not the real problem. 

And since you have that nice function, why not use it BEFORE the call to mount so you can tell them right away that it's an unknown file system? 

Since log.Fatalf exits you don't need an else.

Oh and ... one reason -t may not be mandatory in fiuture is because of bind mounts, which this command does not support but which also don't require a file system name. So -t at some point may go back to being optional. Mount is such fun :-)

Reply:
The code you point to is verification that a -t has been supplied by the user. so that is a check before.

I shall do things differently in the part that actually document the error.
My business space is embedded so boot time is important (you don't want to wait for your rear camera when you start your car and want to backup).
Even this is little work, I don't want checks to slow down regular work programmed in script because I want to help users.
So, as a "philosophy" I prefer to do "post-portem" documentation rather than screening things at the beginning. Except if checks are here to prevent harm of the command to be executed.

If you agree, I'll propose a different way to document the error after it happens.